loki/apps-v1.Deployment-querier.yaml

```diff
@@ -30,7 +29,6 @@ spec:
       containers:
       - name: querier
         args:
-        - -boltdb.shipper.cache-location=/data/boltdb-cache
```

Comment:
I'm not sure if this can safely be removed.

Reply:
According to the documentation, if an index gateway is configured, the Loki querier should use the index gateway for loading indexes. Are there concerns other than this for whether or not this is safe to remove?

https://grafana.com/docs/loki/latest/operations/storage/boltdb-shipper/#index-gateway

Reply:
I'm not sure if the boltdb shipper cache is used for more than just indexes; e.g. does it cache chunks there too?

Reply:
This should be safe to remove on the querier, although it still needs to be present on the index-gateway. It shouldn't _hurt_ having this config on the querier either, it just won't do anything if using the index-gateway.

Reply:
Thanks @owen-d!

Additional question: on the rulers, they still need to be StatefulSets, right? But they no longer need the boltdb cache?
However I there's nothing to cleanup the boltdb-cache folder from their PV? i.e. as a post-upgrade instruction, we should have people run `rm -rf /data/boltdb-cache` on their ruler instances?

Reply:
The rulers can safely be moved back to deployments in conjunction with the index-gateway. Previously, they were run as statefulsets in order to cache the synchronized index state from object storage, but could also be run as deployments too, at the cost of more frequent synchronization across restarts.